### PR TITLE
pixie: init at 0-1333

### PR DESCRIPTION
**Pull request by bendlas**
###### Things done:
- [x] Tested via `nix.useChroot`.
- [x] Built on platform(s): `Linux nitox 4.4.2 #1-NixOS SMP Wed Feb 17 20:31:25 UTC 2016 x86_64 GNU/Linux, 16.03-pre-401d590a64736d3c (Emu)`
- [x] Tested compilation of all pkgs that depend on this change.
- [x] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This is the pixie language, a clojure-like lisp, utilizing the pypy vm toolkit for JIT, GC and C bindings.
see https://github.com/pixie-lang/pixie

**EDIT**
Regarding closure size: The package weighs in at around 6MB, and only depends on `libffi`, `libuv`, `boost`, `zlib` and `gcc`

### CODE REVIEW

**Comment by domenkozar** [comment]
@bendlas couldn't you use pypy that we already have?

**Comment by bendlas** [comment]
That's what I did originally, when the package makefile would pull a pypy snapshot to build against. I encouraged the pypy devs to develop against latest stable and they recognized the problem, but instead chose to fix the pypy version to a specific commit, which I copied.

Note, that pixie doesn't actually build pypy, but it uses their toolchain to translate itself to C and to add GC + JIT

**Comment by bendlas** [comment]
@domenkozar could this still go into 16.03 per chance?

**Comment by domenkozar** [comment]
Sure, let's just wait for travis-ci

**Comment by bendlas** [comment]
Added `dust` the package manager / dev tool for `pixie`

**Comment by joachifm** [comment]
Looks like this is ready to merge?

**Comment by bendlas** [comment]
All good from my part

File: pkgs/development/interpreters/pixie/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      cp pixie-src/pixie-vm $out/share/pixie-vm
+      cp -R pixie-src/pixie $out/share/pixie
+      makeWrapper $out/share/pixie-vm $out/bin/pxi \
+        --prefix LD_LIBRARY_PATH : ${library-path} \
+        --prefix C_INCLUDE_PATH : ${include-path} \
+        --prefix LIBRARY_PATH : ${library-path} \
+        --prefix PATH : ${bin-path}
+    '';
+    meta = {
+      description = "Pixie is a clojure-like lisp, built with the pypy vm toolkit.";
```
Comment (jagajaga) [inline pkgs/development/interpreters/pixie/default.nix]: @zimbatm we omit pkg's name in the description. And also a dot.

Comment (joachifm) [inline pkgs/development/interpreters/pixie/default.nix]: Sorry, I was a bit quick, thought this had passed review. Pushed a fix for it.

**Pull request merged**